File: backend/app/models/restaurant.js
const mongoose = require('mongoose');

const restaurantSchema = new mongoose.Schema({
   name: { type: String, required: true },
   address: String,
   menu: [
      {
         dishName: String,
         price: Number,
         quantityAvailable: Number
      }
   ]
});

module.exports = mongoose.model('Restaurant', restaurantSchema);

File: backend/models/restaurant.py
# FILE: backend/models/restaurant.py
# ============================================================================

from dataclasses import dataclass
from typing import List

@dataclass
class MenuItem:
    item_id: int
    name: str
    price: float
    quantity_available: int

@dataclass
class Restaurant:
    restaurant_id: int
    name: str
    cuisine: str
    menu: List[MenuItem]
    rating: float
    is_active: bool = True